### PR TITLE
Revised implementation of enhancement: add temp reading of fritz box …

### PR DESCRIPTION
**Pull request by mwittig**
…#2, includes fixes for: No values anymore after update #7

### CODE REVIEW

File: fritz.coffee [hunk 1]
```diff
@@ -147,20 +164,76 @@ module.exports = (env) ->
...
       @plugin.fritzCall("getSwitchState", @config.ain)
-        .then (state) =>
-          @_setState(if state then on else off)
-          @plugin.fritzCall("getSwitchPower", @config.ain)
-          .then (power) =>
-            @_setPower(power)
-            @plugin.fritzCall("getSwitchEnergy", @config.ain)
-            .then (energy) =>
-              @_setEnergy(Math.round(energy / 100.0) / 10.0)
+      .then (state) =>
```
Comment (andig) [inline fritz.coffee:174]: why is this updated/ outdented now?

File: fritz.coffee [hunk 2]
```diff
@@ -147,20 +164,76 @@ module.exports = (env) ->
...
+          @plugin.fritzCall("getSwitchEnergy", @config.ain)
+          .then (energy) =>
+            @_setEnergy(Math.round(energy / 100.0) / 10.0)
+      .error (error) ->
+        env.logger.error "Cannot access #{error.options?.url}: #{error.response?.statusCode}"
+
+    # poll device according to interval
+    requestUpdateGetDeviceListInfo:->
+      @plugin.fritzCall("getDeviceListInfo", @config.ain)
+        .then (xmlDeviceInfo) =>
```
Comment (andig) [inline fritz.coffee:188]: any this line is not?

File: fritz.coffee [hunk 3]
```diff
@@ -147,20 +164,76 @@ module.exports = (env) ->
...
-          @plugin.fritzCall("getSwitchPower", @config.ain)
-          .then (power) =>
-            @_setPower(power)
-            @plugin.fritzCall("getSwitchEnergy", @config.ain)
-            .then (energy) =>
-              @_setEnergy(Math.round(energy / 100.0) / 10.0)
+      .then (state) =>
+        @_setState(if state then on else off)
+        @plugin.fritzCall("getSwitchPower", @config.ain)
+        .then (power) =>
```
Comment (andig) [inline fritz.coffee:177]: same as this one?

**Comment by andig** [comment]
Are you confident now that this works for everybody?
Are the fixes from https://github.com/andig/pimatic-fritz/pull/10/files no longer needed?

File: fritz.coffee (hunk 1 above)
Comment (mwittig) [inline fritz.coffee:174]: Sorry, my IDE (IDEA) just did this. It should not have any impact as the line is part of a chained call. Feel free to recover the original ident. Note however, that the following two then() class aren't indented on master, either! See https://github.com/andig/pimatic-fritz/blob/master/fritz.coffee#L156 and follwing three lines! So, indentation on master varies.

File: fritz.coffee (hunk 3 above)
Comment (mwittig) [inline fritz.coffee:177]: This line is not indented in master. See https://github.com/andig/pimatic-fritz/blob/master/fritz.coffee#L156 an followiing three lines!

File: fritz.coffee (hunk 2 above)
Comment (mwittig) [inline fritz.coffee]: No idea.

Comment (mwittig) [inline fritz.coffee:188]: Indentation on master also varies. See comments above. It is probably due to the IDE. I am using IDEA.

**Comment by mwittig** [comment]
I have tested both update modes thoroughly. Indentation changes have no impact on semantics (see comments on diff). Code changes of PR#10 are included in the current PR.

**Comment by andig** [comment]
Could you move the PR10 changes to smartfritz? As it helps abstacting the basic fb api I believe thats were it belongs. Fine to merge then. As for indentation I prefer it unchanged if code is not touched but I realize that IDEs may have differing options.

**Comment by mwittig** [comment]
@andig happy to do that eventually, but I don't have much time right now. So, it'll be great if you can accept the PR as is for now. Regarding smartfritz I am wondering whether or not @nischelwitzer is still maintaining smartfritz. Did you consider contributing your code changes back to the original repo?

**Comment by andig** [comment]
> Regarding smartfritz I am wondering whether or not @nischelwitzer is still maintaining smartfritz. Did you consider contributing your code changes back to the original repo?

Tried to, but never got a response: https://github.com/nischelwitzer/smartfritz/issues/1. As smartfritz-promise has a different api it doesn't make much sense anyway.

> happy to do that eventually, but I don't have much time right now. So, it'll be great if you can accept the PR as is for now.

Same for me... Please move the one function to https://github.com/andig/smartfritz

**Comment by andig** [comment]
ping @mwittig

**Comment by mwittig** [comment]
Ah, ok. Sorry for not looking into this any earlier. Hope, I find some time to get this done next week.

**Comment by andig** [comment]
I've added a `TemperatureSensor` based on the XML api that will be able to read temps of both switches and thermostats. This PR is no longer needed.

**Pull request closed without merging**